qiskit/transpiler/passmanager.py

```diff
@@ -71,6 +71,53 @@ def callback_func(**kwargs):
         self.callback = callback
         self.property_set = None
 
+    def append_best_of(self, passes, property_name, break_condition=None, reverse=None):
```

Comment:
Instead of a whole new method (which is used in very special cases) can you make this a flow_controller? Similar to the existing `do_while` and `condition`. 

So something like:
```
pm.append([BasicSwap(coupling_map), LookaheadSwap(coupling_map), Depth()], best_of=lambda property_set: min(property_set['depth'])])
```

I realize this is a bit awkward, but I like the consistency.

Alternatively, what about special methods for the other flow controllers:
`pm.append_do_while()`
`pm.append_conditional()`

Reply:
What if you want to choose between a sequence of passes, not a single pass?
```
       /--> [pass_A1, pass_A2, pass_A3, analysis_pass]
circuit 
       \--> [pass_B1, pass_B2, pass_B3, analysis_pass]
```

Reply:
I agree consistency of access to the different flow controllers be best from a user's perspective. Maybe something like how we construct the working list in the RunningPassManager, e.g. `pm.append(FlowController(pass_list, **flow_controller_args))`. (And if no `FlowController` is provided, `Linear` is assumed.)

Reply:
Something like the above would also let us having something like `pm.append(BestOfController(pass_set1, pass_set2, ..., pass_setN, property_name='foo'))` which seems less likely to cause confusion than the current structure (and could potentially more easily support more complex structures, like best of over several flow controllers or pass managers) .

e.g if I write
```
pm.append_best_of([pass1, pass2], property_name='foo')
pm.append_best_of([pass3, pass4], property_name='bar')
pm.append_best_of([pass5, pass6], property_name='foo')
```
From having read the code, its clear that  `pass1/pass2` wouldn't be compared against `pass5/pass6` but I could see it being a source of confusion. Whether or not `append_best_of` starts a new best of block or appends to an existing one depends on the state of `pm`.

Reply:
> Maybe something like how we construct the working list in the RunningPassManager, e.g. `pm.append(FlowController(pass_list, **flow_controller_args))`

This was added in https://github.com/Qiskit/qiskit-terra/pull/5772